jdaviz/utils.py

```diff
@@ -385,3 +390,52 @@ def total_masked_first_data(self):
     def __setgluestate__(cls, rec, context):
         masks = {key: context.object(value) for key, value in rec['masks'].items()}
         return cls(masks=masks)
+
+
+def download_uri_to_path(possible_uri, cache=False, local_path=None):
+    """
+    Retrieve data from a URI (or a URL). Return the input if it
```

Comment:
I think this needs to explain that depending on input content, it could go through `astroquery.mast` or `astropy.utils.download_file`, which are two very different beasts. The former `cache` is only a `bool` but the latter isn't (bool or "update", see https://docs.astropy.org/en/stable/api/astropy.utils.data.download_file.html#astropy.utils.data.download_file).

Also should maybe say that local file URI (`file:\\`) is not supported.

And what about Cloud FITS (https://docs.astropy.org/en/stable/io/fits/usage/cloud.html#subsetting-fits-files-hosted-in-amazon-s3-cloud-storage) that has `s3:\\`?

Reply:
I also worry about URI injection but hopefully we can delegate that upstream.

Reply:
I've added notes about this to the docstring. As for S3 – we will need to handle this for Roman, and likely with some Roman-specific optimizations. For now, I'm going to leave these excluded from the URI parser, though they might still work if you use the astropy functions to get a streaming HDU.